core/src/HIP/Kokkos_HIP_IsXnack.cpp

```diff
@@ -0,0 +1,97 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+/*--------------------------------------------------------------------------*/
+
+#ifndef KOKKOS_HIP_ISXNACK_HPP
+#define KOKKOS_HIP_ISXNACK_HPP
```

Comment:
```suggestion
#ifndef KOKKOS_IMPL_PUBLIC_INCLUDE
#define KOKKOS_IMPL_PUBLIC_INCLUDE
```
And I am not sure how this was able to compile before since the include guards clash

Reply:
I am still convinced we don't want an include guard of a hpp file in a cpp file